src/adapter/src/coord/sql.rs

```diff
@@ -139,17 +140,15 @@ impl<S: Append + 'static> Coordinator<S> {
 
     /// Handle removing in-progress transaction state regardless of the end action
     /// of the transaction.
-    pub(crate) async fn clear_transaction(
-        &mut self,
-        session: &mut Session,
-    ) -> TransactionStatus<mz_repr::Timestamp> {
-        let (drop_sinks, txn) = session.clear_transaction();
-        self.drop_compute_sinks(drop_sinks).await;
+    pub(crate) async fn clear_transaction(&mut self, conn_id: &ConnectionId) {
+        if let Some(conn_meta) = self.active_conns.get_mut(conn_id) {
+            let drop_sinks = std::mem::take(&mut conn_meta.drop_sinks);
+            self.drop_compute_sinks(drop_sinks).await;
+        }
```

Comment:
```suggestion
        let conn_meta = self.active_conns.get_mut(conn_id).expect("must exist");
        let drop_sinks = std::mem::take(&mut conn_meta.drop_sinks);
        self.drop_compute_sinks(drop_sinks).await;
```

This can expect conns to always exist right?

Reply:
Yes, good point, just updated this.